src/lib/fetch.ts

```diff
@@ -9,85 +10,99 @@ export interface FetchOptions {
   noResolveJson?: boolean
 }
 
-export type RequestMethodType = 'GET' | 'POST' | 'PUT' | 'PATCH' | 'DELETE'
+export interface FetchParameters {
+  signal?: AbortSignal
+}
```

Comment:
just wondering what is this for and why do we add it to every fetch request?

Reply:
It's a bit like a context in Go - allows us to cancel a request. We have them in `postgrest-js` so I think it makes sense to have them here too

https://developer.mozilla.org/en-US/docs/Web/API/AbortSignal